queries.sql

```diff
@@ -57,14 +65,108 @@ JOIN owners o
 ON a.owner_id = o.id 
 WHERE o.full_name = 'Jennifer Orwell' AND s.name = 'Digimon';
 
+-- List of all animals owned by Dean Winchester that haven't tried to escape
 SELECT a.name, a.escape_attempts, o.full_name
 FROM animals a
 JOIN owners o ON a.owner_id = o.id 
 JOIN species s ON a.species_id = s.id
 WHERE a.escape_attempts = 0 AND o.full_name = 'Dean Winchester';
 
+-- Find who owns the most animals
 SELECT o.full_name, COUNT(a.name) 
 FROM animals a
 JOIN owners o ON a.owner_id = o.id GROUP BY o.full_name
 ORDER BY COUNT(*) DESC
 LIMIT 1;
+
+-- Last animal seen by William Tatcher
+SELECT a.name AS animal_name, v.date_of_visit 
+FROM animals a
+JOIN visits v
+ON a.id = v.animal_id 
+WHERE v.vet_id = 1 
+GROUP BY a.name, v.date_of_visit 
+ORDER BY v.date_of_visit
+DESC LIMIT 1;
+
+-- Number of different animals seen by Stephanie Mendez
+SELECT ve.name, COUNT(vi.animal_id) AS animal_count 
+FROM visits vi
+JOIN vets ve
+ON vi.vet_id = ve.id 
+WHERE ve.name = 'Stephanie Mendez' 
+GROUP BY ve.name;
+
+-- List of all vets and their specialties, including vets with no specialties
+SELECT ve.name AS vet_name, s.name AS species_name
+FROM vets ve
+LEFT JOIN specializations sp
+ON ve.id = sp.vet_id 
+LEFT JOIN species s
+ON s.id = sp.species_id;
+
+-- List of all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
+SELECT ve.name AS vet_name, a.name AS animal_name, vi.date_of_visit
+FROM vets ve
+INNER JOIN visits vi
+ON ve.id = vi.vet_id  
+JOIN animals a
+ON a.id = vi.animal_id 
+WHERE ve.name = 'Stephanie Mendez' 
+AND vi.date_of_visit BETWEEN 'Apr 1, 2020' AND 'Aug 30, 2020' 
+GROUP BY ve.name, a.name, vi.date_of_visit;
+
+-- Animal with the most visits to vets
+SELECT a.name, count(*) 
+FROM animals a
+JOIN visits vi
+ON vi.animal_id = a.id
+GROUP BY a.name
+ORDER BY count
+DESC LIMIT 1;
+
+-- Animal that visited Maisy Smith first
+SELECT a.name, ve.name, vi.date_of_visit 
+FROM animals a
+JOIN visits vi
+ON a.id = vi.animal_id
+JOIN vets ve
+ON vi.vet_id = ve.id
+WHERE ve.name = 'Maisy Smith'
+ORDER BY vi.date_of_visit
+ASC lIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit
+SELECT a.name AS animal_name, ve.name AS vet_name, vi.date_of_visit
+FROM animals a
+JOIN visits vi
+ON a.id = vi.animal_id
+JOIN vets ve
+ON vi.vet_id = ve.id
+ORDER BY vi.date_of_visit
+DESC lIMIT 1;
+
+-- Number of visits with a vet who did not specialize in that animal's species
+SELECT COUNT(*) 
+FROM vets ve 
+JOIN visits vi 
+ON vi.vet_id = ve.id
+JOIN animals a
+ON vi.animal_id = a.id
+JOIN specializations sp 
+ON ve.id = sp.vet_id
+WHERE NOT sp.species_id = a.species_id;
```

Comment:
- [ ] This query returns the wrong result. The query should return the total number of visits by animals to vets who were not specialized in the animal's species. For this, the result should be 12. This is the breakdown:

- William Thatcher who specializes in Pokemon had 2 visits by Digimon
- Jack Harkness who specializes in Digimon had 1 visit by a Pokemon
- Maisy Smith doesn't specialize in any species but had 9 visits. 

This makes a total of _**12**_ visits. Kindly write a query that selects the `visits` table that the `LEFT JOIN` the `animals` table that `LEFT JOIN` the `vets` table on `animals.species_id` that isn't in the `species_id` in the specialization table. 👍 😎

Reply:
@Gambit142 Thank you for the feedback. Am on it 👍